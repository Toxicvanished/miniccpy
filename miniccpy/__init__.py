RHF_MODULES = ["rlccd", "rccd", "rccsd", "left_rccsd", "eomrccsd"]
