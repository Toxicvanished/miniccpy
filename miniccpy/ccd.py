import time
import numpy as np
from miniccpy.energy import ccd_energy, hf_energy, hf_energy_from_fock
from miniccpy.diis import DIIS

from miniccpy.updates import update_t2


def doubles_residual(t2, f, g, o, v):
    """Compute the projection of the CCD Hamiltonian on doubles
        X[a, b, i, j] = < ijab | (H_N exp(T2))_C | 0 >
    """
    # intermediates
    I1_oo = f[o, o] + 0.5 * np.einsum("mnef,efin->mi", g[o, o, v, v], t2, optimize=True)
    I1_vv = f[v, v] - 0.5 * np.einsum("mnef,afmn->ae", g[o, o, v, v], t2, optimize=True)
    I2_voov = g[v, o, o, v] + 0.5 * np.einsum("mnef,afin->amie", g[o, o, v, v], t2, optimize=True)
    I2_oooo = g[o, o, o, o] + 0.5 * np.einsum("mnef,efij->mnij", g[o, o, v, v], t2, optimize=True)

    doubles_res = 0.5 * np.einsum("ae,ebij->abij", I1_vv, t2, optimize=True)
    doubles_res -= 0.5 * np.einsum("mi,abmj->abij", I1_oo, t2, optimize=True)
    doubles_res += np.einsum("amie,ebmj->abij", I2_voov, t2, optimize=True)
    doubles_res += 0.125 * np.einsum("abef,efij->abij", g[v, v, v, v], t2, optimize=True)
    doubles_res += 0.125 * np.einsum("mnij,abmn->abij", I2_oooo, t2, optimize=True)

    doubles_res -= np.transpose(doubles_res, (1, 0, 2, 3))
    doubles_res -= np.transpose(doubles_res, (0, 1, 3, 2))
    doubles_res += g[v, v, o, o]

    return doubles_res


def kernel(fock, g, o, v, maxit, convergence, energy_shift, diis_size, n_start_diis, out_of_core, use_quasi):
    """Solve the CCSD system of nonlinear equations using Jacobi iterations
    with DIIS acceleration. The initial values of the T amplitudes are taken to be 0."""

    eps = np.diagonal(fock)
    n = np.newaxis
    e_abij = 1.0 / (-eps[v, n, n, n] - eps[n, v, n, n] + eps[n, n, o, n] + eps[n, n, n, o] + energy_shift)

    nunocc, nocc = fock[v, o].shape
    n2 = nocc**2 * nunocc**2
    ndim = n2

    diis_engine = DIIS(ndim, diis_size, out_of_core)

    t2 = np.zeros((nunocc, nunocc, nocc, nocc))
    old_energy = ccd_energy(t2, g, o, v)

    print("    ==> CCD amplitude equations <==")
    print("")
    print("     Iter               Energy                 |dE|                 |dT|")
    for idx in range(maxit):

        tic = time.time()

        residual_doubles = doubles_residual(t2, fock, g, o, v)

        res_norm = np.linalg.norm(residual_doubles)

        #t2 = update_t2(t2, residual_doubles, fock, g, o, v, energy_shift, quasi=use_quasi)
        t2 += residual_doubles * e_abij

        current_energy = ccd_energy(t2, g, o, v)
        delta_e = np.abs(old_energy - current_energy)

        if delta_e < convergence and res_norm < convergence:
            break

        if idx >= n_start_diis:
            diis_engine.push( (t2), (residual_doubles), idx) 
        if idx >= diis_size + n_start_diis:
            T_extrap = diis_engine.extrapolate()
            t2 = T_extrap.reshape((nunocc, nunocc, nocc, nocc))

        old_energy = current_energy

        toc = time.time()
        minutes, seconds = divmod(toc - tic, 60)
        print("    {: 5d} {: 20.12f} {: 20.12f} {: 20.12f}    {:.2f}m {:.2f}s".format(idx, current_energy, delta_e, res_norm, minutes, seconds))
    else:
        raise ValueError("CCD iterations did not converge")

    diis_engine.cleanup()
    e_corr = ccd_energy(t2, g, o, v)

    return (np.zeros((nunocc, nocc)), t2), e_corr


