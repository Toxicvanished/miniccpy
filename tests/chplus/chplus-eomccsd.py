# set allow numpy built with MKL to consume more threads for tensordot
import os
os.environ["MKL_NUM_THREADS"] = "{}".format(os.cpu_count() - 1)

from miniccpy.driver import run_scf_gamess, run_cc_calc, run_guess, run_eomcc_calc, get_hbar


fock, g, e_hf, o, v = run_scf_gamess("chplus.FCIDUMP", 6, 26, 0)

T, Ecorr  = run_cc_calc(fock, g, o, v, method='ccsd')

H1, H2 = get_hbar(T, fock, g, o, v, method='ccsd')

R0, omega0 = run_guess(H1, H2, o, v, 100)

R, omega, r0 = run_eomcc_calc(R0, omega0, T, H1, H2, o, v, method='eomccsd', state_index=[18, 33, 36, 43], maxit=200)






