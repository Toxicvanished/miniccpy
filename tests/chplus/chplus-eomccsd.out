   System Information:
   ----------------------------------------------------
     Number of correlated electrons = 6
     Number of correlated orbitals = 52
     Number of frozen orbitals = 0
     Number of occupied orbitals = 6
     Number of unoccupied orbitals = 46

      MO #        Energy (a.u.)      Symmetry    Occupation
  -----------------------------------------------------------
     1(1洧럈)           -11.843607            C1           1.0
     2(1洧량)           -11.843607            C1           1.0
     3(2洧럈)            -1.249272            C1           1.0
     4(2洧량)            -1.249272            C1           1.0
     5(3洧럈)            -0.873783            C1           1.0
     6(3洧량)            -0.873783            C1           1.0
     7(4洧럈)            -0.331309            C1           0.0
     8(4洧량)            -0.331309            C1           0.0
     9(5洧럈)            -0.331309            C1           0.0
    10(5洧량)            -0.331309            C1           0.0
    11(6洧럈)            -0.138889            C1           0.0
    12(6洧량)            -0.138889            C1           0.0
    13(7洧럈)            -0.083035            C1           0.0
    14(7洧량)            -0.083035            C1           0.0
    15(8洧럈)            -0.078658            C1           0.0
    16(8洧량)            -0.078658            C1           0.0
    17(9洧럈)            -0.078658            C1           0.0
    18(9洧량)            -0.078658            C1           0.0
   19(10洧럈)            -0.054692            C1           0.0
   20(10洧량)            -0.054692            C1           0.0
   21(11洧럈)             0.040725            C1           0.0
   22(11洧량)             0.040725            C1           0.0
   23(12洧럈)             0.181434            C1           0.0
   24(12洧량)             0.181434            C1           0.0
   25(13洧럈)             0.197594            C1           0.0
   26(13洧량)             0.197594            C1           0.0
   27(14洧럈)             0.197594            C1           0.0
   28(14洧량)             0.197594            C1           0.0
   29(15洧럈)             0.396658            C1           0.0
   30(15洧량)             0.396658            C1           0.0
   31(16洧럈)             0.893716            C1           0.0
   32(16洧량)             0.893716            C1           0.0
   33(17洧럈)             1.232860            C1           0.0
   34(17洧량)             1.232860            C1           0.0
   35(18洧럈)             1.232860            C1           0.0
   36(18洧량)             1.232860            C1           0.0
   37(19洧럈)             1.425290            C1           0.0
   38(19洧량)             1.425290            C1           0.0
   39(20洧럈)             1.425290            C1           0.0
   40(20洧량)             1.425290            C1           0.0
   41(21洧럈)             1.793354            C1           0.0
   42(21洧량)             1.793354            C1           0.0
   43(22洧럈)             2.139052            C1           0.0
   44(22洧량)             2.139052            C1           0.0
   45(23洧럈)             2.139052            C1           0.0
   46(23洧량)             2.139052            C1           0.0
   47(24洧럈)             2.921837            C1           0.0
   48(24洧량)             2.921837            C1           0.0
   49(25洧럈)             3.596680            C1           0.0
   50(25洧량)             3.596680            C1           0.0
   51(26洧럈)            26.888537            C1           0.0
   52(26洧량)            26.888537            C1           0.0

   Reference Energy = -37.902768183690426

    ==> CCSD amplitude equations <==

     Iter               Energy                 |dE|                 |dT|
        0      -0.084051827545       0.084051827545       2.016437646062    0.00m 0.17s
        1      -0.102478558855       0.018426731310       0.298187212210    0.00m 0.13s
        2      -0.108743954997       0.006265396143       0.116090021036    0.00m 0.13s
        3      -0.111523830414       0.002779875417       0.062130106897    0.00m 0.13s
        4      -0.112948744520       0.001424914105       0.036032216251    0.00m 0.15s
        5      -0.113733321685       0.000784577166       0.021935024326    0.00m 0.14s
        6      -0.114184279646       0.000450957960       0.013904528070    0.00m 0.15s
        7      -0.114451348909       0.000267069263       0.009125691534    0.00m 0.15s
        8      -0.114613417866       0.000162068958       0.006159453795    0.00m 0.14s
        9      -0.114713907537       0.000100489671       0.004245239334    0.00m 0.15s
       10      -0.114901964324       0.000188056786       0.000006615507    0.00m 0.15s
       11      -0.114901980139       0.000000015815       0.000001930649    0.00m 0.16s
       12      -0.114901981053       0.000000000914       0.000000322765    0.00m 0.16s

    CC Correlation Energy:      -0.114901981160

CC calculation completed in 0.00m 2.07s
Initial guess energies:
  Guess root  1  =  0.07315511282581548
  Guess root  2  =  0.07315511282581603
  Guess root  3  =  0.07315511282581767
  Guess root  4  =  0.0731551128258199
  Guess root  5  =  0.07315511282582439
  Guess root  6  =  0.07315511282582524
  Guess root  7  =  0.17441687167078473
  Guess root  8  =  0.17441687167079206
  Guess root  9  =  0.4253191627319489
  Guess root  10  =  0.4253191627319496
  Guess root  11  =  0.4253191627319524
  Guess root  12  =  0.42531916273195625
  Guess root  13  =  0.42531916273195725
  Guess root  14  =  0.42531916273195725
  Guess root  15  =  0.47138678549188
  Guess root  16  =  0.47138678549188834
  Guess root  17  =  0.47138678549189833
  Guess root  18  =  0.5621331961396754
  Guess root  19  =  0.6163205997954601
  Guess root  20  =  0.6163205997954687
  Guess root  21  =  0.6163205997954692
  Guess root  22  =  0.6170133657364099
  Guess root  23  =  0.6170133657364184
  Guess root  24  =  0.6677870008476579
  Guess root  25  =  0.667787000847664
  Guess root  26  =  0.6677870008476656
  Guess root  27  =  0.693497338321967
  Guess root  28  =  0.6934973383219702
  Guess root  29  =  0.6934973383219704
  Guess root  30  =  0.6934973383219719
  Guess root  31  =  0.6934973383219764
  Guess root  32  =  0.6934973383219766
  Guess root  33  =  0.699817703664358
  Guess root  34  =  0.703252137124576
  Guess root  35  =  0.703252137124588
  Guess root  36  =  0.7076111626689596
  Guess root  37  =  0.7752004054911705
  Guess root  38  =  0.7752004054911735
  Guess root  39  =  0.7752004054911792
  Guess root  40  =  0.7814610897964854
  Guess root  41  =  0.7814610897964883
  Guess root  42  =  0.7814610897964948
  Guess root  43  =  0.7830867873198517
  Guess root  44  =  0.7954204412943786
  Guess root  45  =  0.7954204412943787
  Guess root  46  =  0.7954204412943798
  Guess root  47  =  0.7954204412943825
  Guess root  48  =  0.795420441294384
  Guess root  49  =  0.7954204412943893
  Guess root  50  =  0.8057294110432516
  Guess root  51  =  0.8057294110432606
  Guess root  52  =  0.8136468138360895
  Guess root  53  =  0.8136468138360924
  Guess root  54  =  0.8136468138360994
  Guess root  55  =  0.8483556253885703
  Guess root  56  =  0.9299588823836075
  Guess root  57  =  0.9538305952031275
  Guess root  58  =  0.9538305952031308
  Guess root  59  =  0.9538305952031351
  Guess root  60  =  0.990047137780882
  Guess root  61  =  0.9900471377808844
  Guess root  62  =  0.990047137780889
  Guess root  63  =  1.0038726454939428
  Guess root  64  =  1.0249524706840516
  Guess root  65  =  1.0608662132414266
  Guess root  66  =  1.0608662132414384
  Guess root  67  =  1.0608662132414406
  Guess root  68  =  1.0608662132414408
  Guess root  69  =  1.0608662132414413
  Guess root  70  =  1.0608662132414417
  Guess root  71  =  1.066827243970051
  Guess root  72  =  1.0668272439700528
  Guess root  73  =  1.0668272439700583
  Guess root  74  =  1.0767216388028413
  Guess root  75  =  1.0767216388028524
  Guess root  76  =  1.0779889135675986
  Guess root  77  =  1.1405713330852678
  Guess root  78  =  1.1405713330852736
  Guess root  79  =  1.1405713330852787
  Guess root  80  =  1.1484029436593286
  Guess root  81  =  1.1636241742055526
  Guess root  82  =  1.1636241742055526
  Guess root  83  =  1.1636241742055542
  Guess root  84  =  1.1636241742055549
  Guess root  85  =  1.1636241742055558
  Guess root  86  =  1.1636241742055566
  Guess root  87  =  1.1795935547746614
  Guess root  88  =  1.1795935547746708
  Guess root  89  =  1.193245272212548
  Guess root  90  =  1.1932452722125508
  Guess root  91  =  1.193245272212555
  Guess root  92  =  1.2093855387031447
  Guess root  93  =  1.3232468864694844
  Guess root  94  =  1.3232468864694906
  Guess root  95  =  1.3232468864694937
  Guess root  96  =  1.3830304360844226
  Guess root  97  =  1.38303043608443
  Guess root  98  =  1.3830304360844339
  Guess root  99  =  1.3907186094270074
  Guess root  100  =  1.5143929104058014

    ==> EOMCCSD iterations <==

     Iter               Energy                 |dE|                 |dR|
        1       0.562133196140       0.000000000000       0.639773750677    0.00m 0.01s
        2       0.504647652156      -0.057485543984       0.159268794117    0.00m 0.01s
        3       0.500647486803      -0.004000165353       0.091576211319    0.00m 0.01s
        4       0.499202796910      -0.001444689892       0.058755359396    0.00m 0.01s
        5       0.494764562019      -0.004438234892       0.181158045631    0.00m 0.01s
        6       0.500519106883       0.005754544864       0.092519146244    0.00m 0.02s
        7       0.499062985482      -0.001456121401       0.029937709185    0.00m 0.02s
        8       0.499022187292      -0.000040798189       0.008614830346    0.00m 0.02s
        9       0.499054559243       0.000032371950       0.004215050473    0.00m 0.03s
       10       0.499061895563       0.000007336321       0.001608320390    0.00m 0.03s
       11       0.499066220265       0.000004324702       0.000590381606    0.00m 0.03s
       12       0.499068351979       0.000002131713       0.000214486115    0.00m 0.03s
       13       0.499070588220       0.000002236241       0.003003212720    0.00m 0.03s
       14       0.499068467535      -0.000002120685       0.000110608897    0.00m 0.03s
       15       0.499068734002       0.000000266468       0.000033879894    0.00m 0.04s
       16       0.499068736317       0.000000002314       0.000011827703    0.00m 0.04s
       17       0.499068747740       0.000000011423       0.000020714370    0.00m 0.04s
       18       0.499068718427      -0.000000029313       0.000005473843    0.00m 0.04s
       19       0.499068728263       0.000000009837       0.000001904100    0.00m 0.04s
       20       0.499068731148       0.000000002885       0.000000527197    0.00m 0.04s
       21       0.499068731374       0.000000000226       0.000000171540    0.00m 0.05s
       22       0.499068731471       0.000000000097       0.000000062249    0.00m 0.05s

    EOMCC Excitation Energy:       0.499068731471
    r0 =       0.044813101324
    REL =       0.090047888832

EOMCC calculation completed in 0.00m 3.13s
    ==> EOMCCSD iterations <==

     Iter               Energy                 |dE|                 |dR|
        1       0.699817703664       0.000000000000       0.529308477807    0.00m 0.00s
        2       0.654568269751      -0.045249433913       0.103920069544    0.00m 0.01s
        3       0.651340192511      -0.003228077241       0.148553888751    0.00m 0.02s
        4       0.655133160318       0.003792967807       0.077815630224    0.00m 0.01s
        5       0.654108796866      -0.001024363451       0.039786098875    0.00m 0.02s
        6       0.653385048180      -0.000723748687       0.101547862482    0.00m 0.02s
        7       0.653889196430       0.000504148250       0.039263432432    0.00m 0.02s
        8       0.653488226488      -0.000400969942       0.023467364375    0.00m 0.02s
        9       0.653163483179      -0.000324743309       0.026709318197    0.00m 0.03s
       10       0.649814336471      -0.003349146708       0.127093710283    0.00m 0.03s
       11       0.650553532819       0.000739196348       0.146390014020    0.00m 0.03s
       12       0.655112059722       0.004558526903       0.041714092910    0.00m 0.03s
       13       0.654173681317      -0.000938378405       0.041314321035    0.00m 0.03s
       14       0.654402325402       0.000228644085       0.017284030561    0.00m 0.03s
       15       0.654360062415      -0.000042262987       0.011514324408    0.00m 0.03s
       16       0.654417964444       0.000057902029       0.009062556313    0.00m 0.04s
       17       0.654369822705      -0.000048141738       0.005193001329    0.00m 0.04s
       18       0.654383032253       0.000013209548       0.004236967120    0.00m 0.04s
       19       0.654351279160      -0.000031753093       0.014476452748    0.00m 0.04s
       20       0.654392389328       0.000041110168       0.002461224865    0.00m 0.04s
       21       0.654384758018      -0.000007631310       0.001011580221    0.00m 0.05s
       22       0.654390273300       0.000005515282       0.000520759885    0.00m 0.05s
       23       0.654390048845      -0.000000224455       0.000456016331    0.00m 0.05s
       24       0.654388214474      -0.000001834371       0.000208481384    0.00m 0.05s
       25       0.654386359767      -0.000001854707       0.000491780887    0.00m 0.06s
       26       0.654387973543       0.000001613776       0.000233077389    0.00m 0.06s
       27       0.654388331484       0.000000357941       0.000096268658    0.00m 0.06s
       28       0.654388302855      -0.000000028629       0.000086326121    0.00m 0.06s
       29       0.654387666720      -0.000000636134       0.000037319468    0.00m 0.07s
       30       0.654387688902       0.000000022182       0.000073021924    0.00m 0.06s
       31       0.654387712377       0.000000023475       0.000058068936    0.00m 0.06s
       32       0.654387714400       0.000000002023       0.000112499087    0.00m 0.06s
       33       0.654387828902       0.000000114502       0.000062925706    0.00m 0.06s
       34       0.654387839121       0.000000010219       0.000070873188    0.00m 0.07s
       35       0.654387713961      -0.000000125160       0.000019434884    0.00m 0.08s
       36       0.654387751442       0.000000037481       0.000006603736    0.00m 0.07s
       37       0.654387761258       0.000000009816       0.000003265307    0.00m 0.07s
       38       0.654387778076       0.000000016818       0.000007090643    0.00m 0.07s
       39       0.654387782205       0.000000004129       0.000007370556    0.00m 0.07s
       40       0.654387745932      -0.000000036273       0.000008669320    0.00m 0.07s
       41       0.654388109965       0.000000364033       0.000192748890    0.00m 0.08s
       42       0.654387760124      -0.000000349841       0.000009098458    0.00m 0.08s
       43       0.654387763115       0.000000002991       0.000007059503    0.00m 0.08s
       44       0.654387762718      -0.000000000397       0.000002890559    0.00m 0.08s
       45       0.654387753232      -0.000000009486       0.000002359862    0.00m 0.09s
       46       0.654387865229       0.000000111997       0.000027064391    0.00m 0.09s
       47       0.654387764822      -0.000000100407       0.000002552185    0.00m 0.09s
       48       0.654387767119       0.000000002297       0.000002455367    0.00m 0.09s
       49       0.654387745425      -0.000000021694       0.000005231546    0.00m 0.09s
       50       0.654387346468      -0.000000398957       0.000263669050    0.00m 0.09s
       51       0.654387831158       0.000000484691       0.000041444219    0.00m 0.10s
       52       0.654387772965      -0.000000058194       0.000005859021    0.00m 0.10s
       53       0.654387757998      -0.000000014967       0.000003426617    0.00m 0.11s
       54       0.654387755415      -0.000000002583       0.000001380703    0.00m 0.11s
       55       0.654387778809       0.000000023394       0.000013237587    0.00m 0.11s
       56       0.654387759712      -0.000000019097       0.000000846203    0.00m 0.11s
       57       0.654387758017      -0.000000001695       0.000002305930    0.00m 0.11s
       58       0.654387761112       0.000000003095       0.000000717880    0.00m 0.12s
       59       0.654387761236       0.000000000124       0.000000314935    0.00m 0.13s
       60       0.654387762814       0.000000001579       0.000000638805    0.00m 0.13s
       61       0.654387762531      -0.000000000283       0.000000440074    0.00m 0.13s
       62       0.654387760183      -0.000000002348       0.000000554388    0.00m 0.13s
       63       0.654387760702       0.000000000519       0.000000227209    0.00m 0.13s
       64       0.654387760345      -0.000000000357       0.000000268587    0.00m 0.14s
       65       0.654387761218       0.000000000873       0.000000598605    0.00m 0.14s
       66       0.654387760754      -0.000000000464       0.000000183186    0.00m 0.16s
       67       0.654387757547      -0.000000003207       0.000001475871    0.00m 0.15s
       68       0.654387760920       0.000000003373       0.000000160957    0.00m 0.15s
       69       0.654387760331      -0.000000000589       0.000000291632    0.00m 0.15s
       70       0.654387763305       0.000000002974       0.000000539378    0.00m 0.15s
       71       0.654387761332      -0.000000001973       0.000000293644    0.00m 0.15s
       72       0.654387762263       0.000000000931       0.000000640869    0.00m 0.16s
       73       0.654387747034      -0.000000015229       0.000006328583    0.00m 0.18s
       74       0.654387761454       0.000000014420       0.000000601929    0.00m 0.17s
       75       0.654387762216       0.000000000762       0.000001007339    0.00m 0.17s
       76       0.654387760056      -0.000000002160       0.000000864392    0.00m 0.17s
       77       0.654387762171       0.000000002115       0.000000582265    0.00m 0.19s
       78       0.654387762128      -0.000000000043       0.000000579045    0.00m 0.19s
       79       0.654387760911      -0.000000001217       0.000000334907    0.00m 0.18s
       80       0.654387763661       0.000000002750       0.000000813726    0.00m 0.17s
       81       0.654387761043      -0.000000002618       0.000000177554    0.00m 0.20s
       82       0.654387760890      -0.000000000153       0.000000064351    0.00m 0.20s

    EOMCC Excitation Energy:       0.654387760890
    r0 =       0.007407475469
    REL =       0.069423674497

EOMCC calculation completed in 0.00m 20.63s
    ==> EOMCCSD iterations <==

     Iter               Energy                 |dE|                 |dR|
        1       0.707611162669       0.000000000000       0.638236760281    0.00m 0.00s
        2       0.647906875572      -0.059704287097       0.177706245284    0.00m 0.01s
        3       0.638170804905      -0.009736070667       0.195198397286    0.00m 0.01s
        4       0.654718980197       0.016548175292       0.237003597731    0.00m 0.01s
        5       0.642875797036      -0.011843183161       0.101115619327    0.00m 0.03s
        6       0.636617026154      -0.006258770882       0.166985310914    0.00m 0.02s
        7       0.643639181406       0.007022155252       0.159114712562    0.00m 0.02s
        8       0.637393462146      -0.006245719260       0.073593782224    0.00m 0.02s
        9       0.636309250055      -0.001084212091       0.028521645679    0.00m 0.02s
       10       0.635635495407      -0.000673754649       0.367519411597    0.00m 0.03s
       11       0.636383251894       0.000747756488       0.013864169701    0.00m 0.03s
       12       0.636370860656      -0.000012391238       0.020034480504    0.00m 0.03s
       13       0.636434643992       0.000063783336       0.008098420744    0.00m 0.03s
       14       0.636386572082      -0.000048071910       0.002855283492    0.00m 0.03s
       15       0.636382625706      -0.000003946376       0.001945505423    0.00m 0.04s
       16       0.636357106581      -0.000025519126       0.001712836957    0.00m 0.04s
       17       0.636338015303      -0.000019091278       0.001778742188    0.00m 0.04s
       18       0.636339519256       0.000001503953       0.001288122426    0.00m 0.04s
       19       0.636323669427      -0.000015849829       0.002848542839    0.00m 0.04s
       20       0.636330245032       0.000006575604       0.001716656325    0.00m 0.04s
       21       0.636333868099       0.000003623067       0.001059196569    0.00m 0.05s
       22       0.636335820179       0.000001952080       0.000369671474    0.00m 0.05s
       23       0.636337206641       0.000001386462       0.000920759441    0.00m 0.06s
       24       0.636336382712      -0.000000823928       0.000187658637    0.00m 0.06s
       25       0.636334578346      -0.000001804366       0.000095837132    0.00m 0.08s
       26       0.636334775236       0.000000196890       0.000041658452    0.00m 0.07s
       27       0.636334996041       0.000000220804       0.000034528256    0.00m 0.06s
       28       0.636335639811       0.000000643771       0.000305338480    0.00m 0.06s
       29       0.636334824128      -0.000000815683       0.000021068395    0.00m 0.06s
       30       0.636334821519      -0.000000002609       0.000022263476    0.00m 0.07s
       31       0.636334885749       0.000000064230       0.000007381888    0.00m 0.07s
       32       0.636334908587       0.000000022837       0.000002862501    0.00m 0.07s
       33       0.636334844638      -0.000000063949       0.000045123198    0.00m 0.07s
       34       0.636334902181       0.000000057543       0.000001482279    0.00m 0.07s
       35       0.636334904044       0.000000001863       0.000001179590    0.00m 0.07s
       36       0.636334904522       0.000000000478       0.000004989665    0.00m 0.08s
       37       0.636334903869      -0.000000000652       0.000001949135    0.00m 0.08s
       38       0.636334903876       0.000000000006       0.000001694651    0.00m 0.08s
       39       0.636334894215      -0.000000009661       0.000012245749    0.00m 0.08s
       40       0.636334909677       0.000000015463       0.000002610397    0.00m 0.08s
       41       0.636334904011      -0.000000005666       0.000000814345    0.00m 0.08s
       42       0.636334902006      -0.000000002005       0.000000346316    0.00m 0.08s
       43       0.636334903132       0.000000001127       0.000000288516    0.00m 0.09s
       44       0.636334902847      -0.000000000286       0.000000541161    0.00m 0.09s
       45       0.636334902856       0.000000000009       0.000000517124    0.00m 0.09s
       46       0.636334903274       0.000000000418       0.000000425625    0.00m 0.09s
       47       0.636334903038      -0.000000000236       0.000000162972    0.00m 0.09s
       48       0.636334902623      -0.000000000415       0.000000152642    0.00m 0.09s
       49       0.636334902658       0.000000000035       0.000000110514    0.00m 0.09s
       50       0.636334902864       0.000000000205       0.000000040526    0.00m 0.10s

    EOMCC Excitation Energy:       0.636334902864
    r0 =       0.058624100573
    REL =       0.119163731842

EOMCC calculation completed in 0.00m 9.64s
    ==> EOMCCSD iterations <==

     Iter               Energy                 |dE|                 |dR|
        1       0.783086787320      -0.000000000000       0.525651023644    0.00m 0.00s
        2       0.736580928564      -0.046505858756       0.091363769197    0.00m 0.01s
        3       0.734750743633      -0.001830184931       0.086084773807    0.00m 0.01s
        4       0.736256262238       0.001505518605       0.076609042612    0.00m 0.01s
        5       0.738845305174       0.002589042936       0.187105370623    0.00m 0.02s
        6       0.735910584781      -0.002934720393       0.040125163250    0.00m 0.02s
        7       0.733767276853      -0.002143307928       0.123775415038    0.00m 0.02s
        8       0.736783653900       0.003016377047       0.040191546220    0.00m 0.02s
        9       0.735814601181      -0.000969052719       0.047000948095    0.00m 0.03s
       10       0.729523280220      -0.006291320961       0.165871871203    0.00m 0.03s
       11       0.737045026550       0.007521746330       0.071249566261    0.00m 0.03s
       12       0.733812038097      -0.003232988453       0.062082747865    0.00m 0.03s
       13       0.731623928822      -0.002188109276       0.057066745136    0.00m 0.03s
       14       0.730707631967      -0.000916296854       0.035416699947    0.00m 0.03s
       15       0.730304012424      -0.000403619543       0.030666393471    0.00m 0.04s
       16       0.730924653551       0.000620641127       0.087694385837    0.00m 0.04s
       17       0.729579823973      -0.001344829577       0.018752917919    0.00m 0.04s
       18       0.729646584535       0.000066760561       0.034068050374    0.00m 0.04s
       19       0.729600203610      -0.000046380925       0.009552855082    0.00m 0.04s
       20       0.729613011825       0.000012808214       0.018746926569    0.00m 0.05s
       21       0.729567903104      -0.000045108721       0.005139714830    0.00m 0.05s
       22       0.729535107141      -0.000032795963       0.003300712357    0.00m 0.05s
       23       0.729573818936       0.000038711795       0.002144948972    0.00m 0.05s
       24       0.729515118533      -0.000058700403       0.009749021574    0.00m 0.05s
       25       0.729566451640       0.000051333107       0.001749896141    0.00m 0.06s
       26       0.729526614586      -0.000039837055       0.001696837536    0.00m 0.06s
       27       0.729542028653       0.000015414067       0.001280316260    0.00m 0.07s
       28       0.729544009115       0.000001980462       0.000626788196    0.00m 0.07s
       29       0.729543850398      -0.000000158717       0.001155189510    0.00m 0.07s
       30       0.729545434444       0.000001584045       0.001031424906    0.00m 0.07s
       31       0.729542163732      -0.000003270711       0.000611424183    0.00m 0.07s
       32       0.729545944487       0.000003780754       0.000220054059    0.00m 0.07s
       33       0.729546991038       0.000001046552       0.000159518540    0.00m 0.07s
       34       0.729545905140      -0.000001085899       0.000203176881    0.00m 0.14s
       35       0.729546570284       0.000000665145       0.000053801742    0.00m 0.08s
       36       0.729546072534      -0.000000497751       0.000154490819    0.00m 0.09s
       37       0.729546865558       0.000000793025       0.000024681593    0.00m 0.08s
       38       0.729546924747       0.000000059189       0.000031060713    0.00m 0.08s
       39       0.729546871663      -0.000000053085       0.000015050626    0.00m 0.09s
       40       0.729546871115      -0.000000000548       0.000020066483    0.00m 0.09s
       41       0.729546882781       0.000000011666       0.000006222877    0.00m 0.08s
       42       0.729546921324       0.000000038542       0.000013645806    0.00m 0.09s
       43       0.729546865609      -0.000000055715       0.000010218657    0.00m 0.10s
       44       0.729547082879       0.000000217271       0.000054066556    0.00m 0.10s
       45       0.729546882694      -0.000000200185       0.000007624661    0.00m 0.09s
       46       0.729546881392      -0.000000001302       0.000005997761    0.00m 0.09s
       47       0.729546900580       0.000000019188       0.000002646604    0.00m 0.09s
       48       0.729546906660       0.000000006080       0.000029791825    0.00m 0.09s
       49       0.729546892948      -0.000000013712       0.000001858179    0.00m 0.09s
       50       0.729546899484       0.000000006536       0.000000651601    0.00m 0.10s
       51       0.729546899128      -0.000000000357       0.000000700923    0.00m 0.11s
       52       0.729546900183       0.000000001055       0.000000519903    0.00m 0.11s
       53       0.729546860278      -0.000000039904       0.000009416966    0.00m 0.12s
       54       0.729546896930       0.000000036652       0.000000352199    0.00m 0.11s
       55       0.729546899164       0.000000002234       0.000000495227    0.00m 0.12s
       56       0.729546894424      -0.000000004740       0.000001709933    0.00m 0.12s
       57       0.729546900145       0.000000005721       0.000000471965    0.00m 0.13s
       58       0.729546900424       0.000000000279       0.000000561211    0.00m 0.12s
       59       0.729546897254      -0.000000003170       0.000000395707    0.00m 0.12s
       60       0.729546897876       0.000000000622       0.000000287580    0.00m 0.12s
       61       0.729546898744       0.000000000868       0.000000487509    0.00m 0.12s
       62       0.729546898206      -0.000000000539       0.000000218501    0.00m 0.12s
       63       0.729546898293       0.000000000087       0.000000175545    0.00m 0.12s
       64       0.729546898155      -0.000000000138       0.000000155527    0.00m 0.13s
       65       0.729546898408       0.000000000253       0.000000067064    0.00m 0.13s

    EOMCC Excitation Energy:       0.729546898408
    r0 =       0.055343861882
    REL =       0.108352118884

EOMCC calculation completed in 0.00m 14.61s
